jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ConditionalHandler.java

```diff
@@ -0,0 +1,738 @@
+
+    private boolean checkPredicates(Request request)
+    {
+        return _predicates.test(request);
+    }
+
+    @Override
+    protected void doStart() throws Exception
+    {
+        Handler next = getHandler();
+
+        MethodHandles.Lookup lookup = MethodHandles.lookup();
+        MethodType handleType = methodType(boolean.class, Request.class, Response.class, Callback.class);
+
+        // Determine if this classes doHandle method has been extended
+        boolean doHandleExtended;
+        try
+        {
+            doHandleExtended = getClass().getDeclaredMethod("doHandle", Request.class, Response.class, Callback.class).getDeclaringClass() != ConditionalHandler.class;
+        }
+        catch (NoSuchMethodException e)
+        {
+            doHandleExtended = false;
+        }
+
+        // Determine the handling if the conditions are met.
+        // If doHandle has been extended (or there is no next handler), then we invoke the method on this class,
+        // otherwise we can directly invoke the handle method on the next class
+        _doHandle = doHandleExtended || next == null
+            ? lookup.findVirtual(ConditionalHandler.class, "doHandle", handleType).bindTo(this)
+            : lookup.findVirtual(next.getClass(), "handle", handleType).bindTo(next);
+
+        // Determine the handling if the conditions are not met
+        MethodHandle doNotHandle = switch (_notHandled)
+        {
+            case DO_NOT_HANDLE ->
+                // Invoke the doNotHandle method
+                lookup.findVirtual(ConditionalHandler.class, "doNotHandle", handleType).bindTo(this);
+            case SKIP_THIS ->
+            {
+                // Skip the doNotHandle method of this handler and call handle of the next handler
+                if (next != null)
+                {
+                    MethodHandle nextHandle = lookup.findVirtual(next.getClass(), "handle", handleType);
+                    yield nextHandle.bindTo(next);
+                }
+                throw new IllegalStateException("No next Handler");
+            }
+            case SKIP_NEXT ->
+            {
+                // Skip the handle method of the next handler and call its next handler.
+                if (doHandleExtended)
+                    throw new IllegalStateException("doHandle method is extended in SKIP_NEXT mode");
+                if (getHandler() instanceof Handler.Singleton nextWrapper && nextWrapper.getHandler() != null)
+                {
+                    Handler nextNextHandler = nextWrapper.getHandler();
+                    MethodHandle nextHandle = lookup.findVirtual(nextNextHandler.getClass(), "handle", handleType);
+                    yield nextHandle.bindTo(nextNextHandler);
+                }
+                throw new IllegalStateException("No next Handler.Wrapper next");
+            }
+        };
+
+        // If we have method conditions, guard the doHandle method with checkMethod
+        if (!_methods.isEmpty())
```

Comment:
This is 2 `Set.isEmpty()` checks, easily inlined and predicted... Not sure it's worth all the `MethodHandle` effort rather than just calling it every time.
Code clarity would benefit much.